lib/class-wp-rest-meta-fields.php

```diff
@@ -0,0 +1,285 @@
+	 * @return mixed Prepared value.
+	 */
+	protected function prepare_value_for_response( $value, $name, $args ) {
+		switch ( $args['schema']['type'] ) {
+			case 'string':
+				$value = strval( $value );
+				break;
+			case 'number':
+				$value = floatval( $value );
+				break;
+			case 'boolean':
+				$value = (bool) $value;
+				break;
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Update settings for the settings object.
+	 *
+	 * @param  WP_REST_Request $request Full detail about the request.
+	 * @return WP_Error|array
+	 */
+	public function update_value( $params, $data, $field_name, $request ) {
+		$options = $this->get_registered_fields();
+
+		foreach ( $options as $name => $args ) {
+			if ( ! array_key_exists( $name, $params ) ) {
+				continue;
+			}
+
+			// a null value means reset the option, which is essentially deleting it
+			// from the database and then relying on the default value.
+			if ( is_null( $params[ $name ] ) ) {
+				$result = $this->delete_meta_value( $request['id'], $name );
+			} elseif ( $args['single'] ) {
+				$result = $this->update_meta_value( $request['id'], $name, $params[ $name ] );
+			} else {
+				$result = $this->update_multi_meta_value( $request['id'], $name, $params[ $name ] );
+			}
+
+			if ( is_wp_error( $result ) ) {
+				return $result;
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Delete meta value for an object.
+	 *
+	 * @param int $object Object ID the field belongs to.
+	 * @param string $name Key for the field.
+	 * @return bool|WP_Error True if meta field is deleted, error otherwise.
+	 */
+	protected function delete_meta_value( $object, $name ) {
+		if ( ! current_user_can( 'delete_post_meta', $object, $name ) ) {
+			return new WP_Error(
+				'rest_cannot_delete',
+				sprintf( __( 'You do not have permission to edit the %s custom field.' ), $name ),
+				array( 'key' => $name, 'status' => rest_authorization_required_code() )
+			);
+		}
+
+		if ( ! delete_metadata( $this->get_meta_type(), $object, wp_slash( $name ) ) ) {
+			return new WP_Error(
+				'rest_meta_database_error',
+				__( 'Could not delete meta value from database.' ),
+				array( 'key' => $name, 'status' => WP_HTTP::INTERNAL_SERVER_ERROR )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Update multiple meta values for an object.
+	 *
+	 * Alters the list of values in the database to match the list of provided values.
+	 *
+	 * @param int $object Object ID.
+	 * @param string $name Key for the custom field.
+	 * @param array $values List of values to update to.
+	 * @return bool|WP_Error True if meta fields are updated, error otherwise.
+	 */
+	protected function update_multi_meta_value( $object, $name, $values ) {
+		if ( ! current_user_can( 'edit_post_meta', $object, $name ) ) {
+			return new WP_Error(
+				'rest_cannot_update',
+				sprintf( __( 'You do not have permission to edit the %s custom field.' ), $name ),
+				array( 'key' => $name, 'status' => rest_authorization_required_code() )
+			);
+		}
+
```

Comment:
I did just run into a problem, I think a lot of devs could run into. When I haven't registered the meta as `single` and want to update, I end up here. Lets say I run a `POST` request like this: http://example.com/wp-json/wp/v2/posts/1177?meta[test]=just-a-string
`$values` would be a string here, but needs to be an array. My workaround was on L170:

```
		if ( ! is_array( $values ) ) {
			$values = array( $values );
		}
```